### PR TITLE
Allow modules with case list menu item but no forms

### PR DESCRIPTION
**Pull request by NoahCarnahan**
Allows user to build apps that contain modules with a case list menu item set to show but no forms.
Addresses http://manage.dimagi.com/default.asp?157080

@dannyroberts

### CODE REVIEW

File: corehq/apps/app_manager/models.py [hunk 1]
```diff
@@ -1732,6 +1727,11 @@ def case_list_filter(self):
 
     def validate_for_build(self):
         errors = super(Module, self).validate_for_build()
+        if not self.forms and not self.case_list.show:
```
Comment (NoahCarnahan) [inline corehq/apps/app_manager/models.py]: Should I also allow building if `referral_list.show` or `task_list.show` are `True`? I don't know exactly how these things are handled, but it looks like they are a little different...

File: corehq/apps/app_manager/models.py [hunk 2]
```diff
@@ -2609,6 +2613,15 @@ def get_case_errors(self, needs_case_type, needs_case_detail, needs_referral_det
             for error in errors:
                 yield error
 
+    def validate_for_build(self):
```
Comment (NoahCarnahan) [inline corehq/apps/app_manager/models.py]: This hasn't been tested as I'm not sure how to make a careplan app. Can anyone point me in the right direction?

File: corehq/apps/app_manager/models.py (hunk 1 above)
Comment (snopoke) [inline corehq/apps/app_manager/models.py]: No, those are legacy and hopefully we'll be dropping support for then soon.

Comment (NoahCarnahan) [inline corehq/apps/app_manager/models.py]: Good to know, thanks.

File: corehq/apps/app_manager/models.py (hunk 2 above)
Comment (snopoke) [inline corehq/apps/app_manager/models.py]: If you enable the feature flag you'll get a drop down menu when adding a module and one of the options will be care plan.

Comment (NoahCarnahan) [inline corehq/apps/app_manager/models.py]: Ah I should have found that, thanks.

**Comment by NoahCarnahan** [comment]
Ok, I've tested the careplan component. This is good to go.

**Pull request merged**